### PR TITLE
ollama: fix a symlink and use a smaller rocm path

### PR DESCRIPTION
**Pull request by abysssol**
## Description of changes

Minor improvements that are not urgent. Hopefully a couple people can test ollama with rocm enabled to ensure this doesn't introduce regressions.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/ollama/default.nix [hunk 1]
```diff
@@ -68,6 +68,10 @@ let
       cudaPackages.cudatoolkit
       cudaPackages.cuda_cudart
     ];
+    postBuild = ''
+      rm "$out/lib64"
```
Comment (drupol) [inline pkgs/tools/misc/ollama/default.nix:72]: If it's a directory, shouldn't we use `-f` or eventually `-rf` ?

Comment (abysssol) [reply pkgs/tools/misc/ollama/default.nix:72]: it's a symlink

**Comment by abysssol** [comment]
@drupol you didn't think waiting for third-party testing was necessary?

**Comment by drupol** [comment]
Tests on Darwin take ages. I usually merge when x86's are green. Especially for such a minor change.

**Comment by abysssol** [comment]
I meant other people testing that ollama correctly uses rocm on their machine (instead of falling back to cpu), which can't practically be tested in the cloud.

if you mean `ollama.passthru.tests`, that should have errored on all systems if the nixos tests ran. `nix-build -A ollama.passthru.tests` correctly returns an error when building #296309 locally, but ofborg was all green. So I don't think it's running the nixos service test.

**Comment by ulrikstrid** [comment]
After this was merged to master I get this when running `ollama serve`
```
time=2024-03-25T08:03:30.546+01:00 level=INFO source=images.go:806 msg="total blobs: 11"
time=2024-03-25T08:03:30.547+01:00 level=INFO source=images.go:813 msg="total unused blobs removed: 0"
time=2024-03-25T08:03:30.547+01:00 level=INFO source=routes.go:1110 msg="Listening on 127.0.0.1:11434 (version 0.1.29)"
time=2024-03-25T08:03:30.547+01:00 level=INFO source=payload_common.go:112 msg="Extracting dynamic libraries to /tmp/ollama367513290/runners ..."
time=2024-03-25T08:03:30.567+01:00 level=INFO source=payload_common.go:139 msg="Dynamic LLM libraries [cpu cpu_avx2 cpu_avx]"
time=2024-03-25T08:03:30.567+01:00 level=INFO source=gpu.go:77 msg="Detecting GPU type"
time=2024-03-25T08:03:30.567+01:00 level=INFO source=gpu.go:191 msg="Searching for GPU management library libnvidia-ml.so"
time=2024-03-25T08:03:30.567+01:00 level=INFO source=gpu.go:237 msg="Discovered GPU libraries: []"
time=2024-03-25T08:03:30.567+01:00 level=INFO source=cpu_common.go:11 msg="CPU has AVX2"
time=2024-03-25T08:03:30.567+01:00 level=WARN source=amd_linux.go:53 msg="ollama recommends running the https://www.amd.com/en/support/linux-drivers: amdgpu version file missing: /sys/module/amdgpu/version stat /sys/module/amdgpu/version: no such file or directory"
time=2024-03-25T08:03:30.568+01:00 level=INFO source=amd_linux.go:88 msg="detected amdgpu versions [gfx1101]"
time=2024-03-25T08:03:30.568+01:00 level=WARN source=amd_linux.go:350 msg="amdgpu detected, but no compatible rocm library found.  Either install rocm v6, or follow manual install instructions at https://github.com/ollama/ollama/blob/main/docs/linux.md#manual-install"
time=2024-03-25T08:03:30.568+01:00 level=WARN source=amd_linux.go:99 msg="unable to verify rocm library, will use cpu: no suitable rocm found, falling back to CPU"
time=2024-03-25T08:03:30.568+01:00 level=INFO source=routes.go:1133 msg="no GPU detected"
```
So this does not work for me at least

**Comment by abysssol** [comment]
@ulrikstrid did you remeber to enable `acceleration = "rocm";`?

**Comment by ulrikstrid** [comment]
> @ulrikstrid did you remeber to enable `acceleration = "rocm";`? Those logs look like mine when not compiled with rocm support.

You are correct, don't test these things too early in the morning kids. I'm currently building rocsolver so I'll report back when that is done.

**Comment by ulrikstrid** [comment]
Yes, this works as expected. Sorry for the noise.

**Pull request merged**